### PR TITLE
ADD: Arduino environment/analog readout hardware layer

### PR DESCRIPTION
**Pull request by mjmucha**
An Arduino firmware based on the NTC readout firmware has been added to read out the pure analog values. This firmware helps to standardize experimental setups with multiple Arduino readout boards by moving the temperature/humidity/pressure calculation to the 'EnvironmentReadout' class instances in the hardware drivers list. It can also reduce the number of different Arduino firmwares required for larger setups to one.
This step was also necessary for temperature-corrected humidity measurements in more extreme measurement ranges.

The `EnvironmentReadout` class has readout functions for temperature, humidity and pressure. The calibration/conversion parameters can be set as init parameters of the corresponding hardware driver, as follows:

```
hw_drivers:
  - name      : ReadoutEnvHumPres
    type      : arduino_env_readout
    interface : ArduinoBoardEnv
    init :
      ntc_limits : [-55, 70]
      resistors: 51100
      pressure_pin: 7
      humidity_pin: 6
      humidity_params:
        slope: 0.031483
        offset: 0.826
      humidity_temp_correction_params:
        slope: -0.00216
        offset: 1.0546
      pressure_params:
        slope: 27.6
        offset: 33.45
```

TL;DR: These changes simplify the readout of multiple Arduino "Environment" boards by moving the hardware-dependent calibration constants to the hardware driver configuration as an init parameter. The conversion from analog readout to temperature/humidity/pressure measurement is done by the hardware driver.

Optional improvements:
- [  ] Allow multiple humidity/pressure pins

### CODE REVIEW

**Comment by YannickDieter** [comment]
Please rebase to check if pipeline is now working.

File: basil/HL/arduino_env_readout.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            "slope": 0.0,
+            "offset": 0.0,
+        }))
+
+        self.pressure_params = dict(self._init.get('pressure_params', {
+            "slope": 0.0,
+            "offset": 0.0,
+        }))
+
+    def setFixedResistance(self, pin, resistance):
```
Comment (cbespin) [inline basil/HL/arduino_env_readout.py:51]: Please use python-style snake_case function names

**Review by cbespin — CHANGES_REQUESTED** [review]
I have not enough experience with arduino to comment on the ino files, but if it works, I trust you. Nicely commented for those who want to contribute!
Only thing is the python-style function naming ;)

**Comment by YannickDieter** [comment]
ping
